src/dotnet/commands/dotnet-pack/PackageGenerator.cs

```diff
@@ -94,6 +94,7 @@ protected virtual void ProcessContext(ProjectContext context)
 
             TryAddOutputFile(context, inputFolder, outputName);
             TryAddOutputFile(context, inputFolder, $"{Project.Name}.xml");
+            TryAddOutputFile(context, inputFolder, $"{Project.Name}.runtimeconfig.json");
```

Comment:
cc @piotrpMSFT @anurse 
Picking up runtimeconfig.json in pack if it is present to solve the dilemma with packing Portable Apps.
Deps Json is generated after restore.